bonfire/bonfire.py

```diff
@@ -1016,6 +1095,190 @@ def _cmd_apps_what_depends_on(
+        if res:
+            raise FatalError(
+                f"Reservation with name {name} already exists"
+            )
+
+        res_config = process_reservation(name, requester, duration)
+
+        log.debug("processed reservation:\n%s", res_config)
+
+        if not bot:
+            if check_for_existing_reservation(res_config["items"][0]["spec"]["requester"]):
+                _warn_of_existing(res_config["items"][0]["spec"]["requester"])
+
+        try:
+            res_name = res_config["items"][0]["metadata"]["name"]
+        except (KeyError, IndexError):
+            raise Exception(
+                "error parsing name of Reservation from processed template, "
+                "check Reservation template"
+            )
+
+        apply_config(None, list_resource=res_config)
+
+        ns_name = wait_on_reservation(res_name, timeout)
+    except KeyboardInterrupt as err:
+        log.error("aborted by keyboard interrupt!")
+        _err_handler(err)
+    except TimedOutError as err:
+        log.error("hit timeout error: %s", err)
+        _err_handler(err)
+    except FatalError as err:
+        log.error("hit fatal error: %s", err)
+        _err_handler(err)
+    except Exception as err:
+        log.exception("hit unexpected error!")
+        _err_handler(err)
+    else:
+        log.info(
+            "namespace '%s' is reserved by '%s' for '%s'",
+            ns_name,
+            res_config["items"][0]["spec"]["requester"],
+            duration,
+        )
+        click.echo(ns_name)
+
+
+@reservation.command("extend")
+@options(_reservation_lookup_options)
+@options(_reservation_process_options)
+def _extend_reservation(name, namespace, requester, duration):
+    def _err_handler(err):
+        msg = f"reservation extension failed: {str(err)}"
+        _error(msg)
+
+    try:
+        res = get_reservation(name, namespace, requester)
+        if res:
+            res_config = process_reservation(
+                res["metadata"]["name"],
+                res["spec"]["requester"],
+                duration,
+            )
+
+            log.debug("processed reservation:\n%s", res_config)
+
+            apply_config(None, list_resource=res_config)
+        else:
+            raise FatalError("Reservation lookup failed")
+    except KeyboardInterrupt as err:
+        log.error("aborted by keyboard interrupt!")
+        _err_handler(err)
+    except TimedOutError as err:
+        log.error("hit timeout error: %s", err)
+        _err_handler(err)
+    except FatalError as err:
+        log.error("hit fatal error: %s", err)
+        _err_handler(err)
+    except Exception as err:
+        log.exception("hit unexpected error!")
+        _err_handler(err)
+    else:
+        log.info(
+            "reservation '%s' extended by '%s'", res["metadata"]["name"], duration
+        )
+
+
+@reservation.command("delete")
+@options(_reservation_lookup_options)
+def _delete_reservation(name, namespace, requester):
```

Comment:
Hit a bug on deleting by ns, reservation name didn't get passed into the `oc delete reservation` command:

```
❯ bonfire reservation delete -n ephemeral-jqacy6
Deleting your reservation will also delete the associated namespace. Proceed? [y/N]: y
2021-09-29 12:13:06 [    INFO] [          MainThread] deleting reservation 'None'
2021-09-29 12:13:06 [    INFO] [          MainThread] running (pid 174234): oc delete reservation 
2021-09-29 12:13:06 [    INFO] [          pid-174234]  |stderr| error: resource(s) were provided, but no name, label selector, or --all flag specified
2021-09-29 12:13:06 [   ERROR] [          MainThread] hit unexpected error!
Traceback (most recent call last):
  File "/home/bsquizza/repos/bonfire/bonfire/bonfire.py", line 1213, in _delete_reservation
    oc("delete", "reservation", name)
  File "/home/bsquizza/repos/bonfire/bonfire/openshift.py", line 187, in oc
    return _exec_oc(*args, **kwargs)
  File "/home/bsquizza/repos/bonfire/bonfire/openshift.py", line 161, in _exec_oc
    raise last_err
  File "/home/bsquizza/repos/bonfire/bonfire/openshift.py", line 124, in _exec_oc
    return cmd.wait()
  File "/home/bsquizza/repos/bonfire/.venv/lib64/python3.9/site-packages/sh.py", line 879, in wait
    self.handle_command_exit_code(exit_code)
  File "/home/bsquizza/repos/bonfire/.venv/lib64/python3.9/site-packages/sh.py", line 905, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /usr/bin/oc delete reservation

  STDOUT:


  STDERR:
error: resource(s) were provided, but no name, label selector, or --all flag specified

ERROR: reservation deletion failed: 

  RAN: /usr/bin/oc delete reservation

  STDOUT:


  STDERR:
error: resource(s) were provided, but no name, label selector, or --all flag specified
```

Reply:
Same happens trying to delete by requester name too. Deleting by reservation `--name` works though.

Reply:
Yeah this is my bad. Fixed the issue in extend and forgot to apply the same fix to delete.

Reply:
Still not quite right

```
❯ bonfire reservation delete -n ephemeral-qvbjz5
Deleting your reservation will also delete the associated namespace. Proceed? [y/N]: y
2021-09-30 15:35:15 [    INFO] [          MainThread] deleting reservation 'bonfire-reservation-27a266c8'
2021-09-30 15:35:15 [    INFO] [          MainThread] running (pid 460366): oc delete reservation bonfire-reservation-27a266c8 
2021-09-30 15:35:16 [    INFO] [          pid-460366]  |stdout| namespacereservation.cloud.redhat.com "bonfire-reservation-27a266c8" deleted
2021-09-30 15:35:16 [    INFO] [          MainThread] reservation 'None' deleted
```

Reply:
Guess really its just that final log msg. Lemme look at it